extensions/Worker.Extensions.SignalRService/src/Hubs/ServerlessHubOfT.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+using Azure.Core.Serialization;
+using Microsoft.AspNetCore.SignalR;
+using Microsoft.Azure.SignalR.Management;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.Azure.Functions.Worker.SignalRService
+{
+    public abstract class ServerlessHub<T> : ServerlessHub where T : class
```

Comment:
Who is meant to derive from this type and `ServerlessHub`? If it is customers, it won't be unit testable since they will never be able to setup `IServiceProvider` correctly with `HubContextProvider` (since it is internal).

Reply:
It is customers to derive from this type. Now the `HubContextProvider` is optional, and they can override those properties such as `Clients`, `GroupManager` for unit tests.

Reply:
Okay - if customers are to derive from this type, I recommend (if you haven't already) you to see what the experience is yourself. Particularly around how easy is it to create a derived type and then unit test that type. Particularly without having access to any of the `internal` types.